### PR TITLE
Restore windows build to default Fedora wxWidgets version

### PR DESCRIPTION
**Pull request by drowe67**
Had some cross compiling issues when trying to build for Windows using wxWidgets 3.1 from https://github.com/drowe67/freedv-gui/pull/68:

```
fdv_win_fed30_c     | [ 62%] Performing configure step for 'wxWidgets'
fdv_win_fed30_c     | cd /home/build/freedv-gui/build_win64/wxWidgets-prefix/src/wxWidgets && ./configure --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --disable-shared --prefix=/home/build/freedv-gui/build_win64/external/dist
fdv_win_fed30_c     | checking build system type... x86_64-w64-mingw32
fdv_win_fed30_c     | checking host system type... x86_64-w64-mingw32
fdv_win_fed30_c     | checking for toolkit... msw
fdv_win_fed30_c     | checking for x86_64-w64-mingw32-gcc... x86_64-w64-mingw32-gcc
fdv_win_fed30_c     | checking whether the C compiler works... yes
fdv_win_fed30_c     | checking for C compiler default output file name... a.exe
fdv_win_fed30_c     | checking for suffix of executables... .exe
fdv_win_fed30_c     | checking whether we are cross compiling... configure: error: in `/home/build/freedv-gui/build_win64/wxWidgets-prefix/src/wxWidgets':
fdv_win_fed30_c     | configure: error: cannot run C compiled programs.
fdv_win_fed30_c     | If you meant to cross compile, use `--host'.
```

So this PR restores the default Fedora wxWidgets version, in order to get Windows cross compiling working again.

### CODE REVIEW

File: build_windows.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ make VERBOSE=1
 # Finally, build freedv-gui
 cd $FREEDVGUIDIR && git pull
 mkdir -p $BUILD_DIR && cd $BUILD_DIR && rm -Rf *
-$CMAKE -DCMAKE_BUILD_TYPE=Debug -DCODEC2_BUILD_DIR=$CODEC2DIR/$BUILD_DIR -DLPCNET_BUILD_DIR=$LPCNETDIR/$BUILD_DIR -DBOOTSTRAP_WXWIDGETS=1 ..
+$CMAKE -DCMAKE_BUILD_TYPE=Debug -DCODEC2_BUILD_DIR=$CODEC2DIR/$BUILD_DIR -DLPCNET_BUILD_DIR=$LPCNETDIR/$BUILD_DIR ..
 make VERBOSE=1
```
Comment (drowe67) [inline build_windows.sh:51]: @hobbes1069 - this caused the Windows build to choke.  If you are keen, pls feel free to fix, however I'm content to build against the default Fedora wxWidgets for now.

Comment (hobbes1069) [reply build_windows.sh:51]: Definitely something broken. I just got back from a Boy Scout canoe trip. May not be able to fix tonight but I will investigate.

**Comment by hobbes1069** [comment]
The wxWindows 3.1.4 installer seems rather buggy. I had to change the $BUILD option to even get it to "build" but then in completed in like two steps, and once I ran cmake again, it couldn't find the libraries. 
```
-- wx-config: /home/build/build/freedv/external/dist/bin/wx-config
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find wxWidgets (missing: wxWidgets_LIBRARIES)
```

**Comment by drowe67** [comment]
@hobbes1069 - thanks for taking a look at it and I hope that Scout camp went well :tent: 

Take your time, we can leave this PR open for now,

**Comment by hobbes1069** [comment]
Ok, I'm trying to tickle some brain cells I haven't exercised in a while... Do we need wxrc at all? One problem with cross compiling is that the binaries are built for Windows not Linux. wx-config is a bash script so it works fine but wxrc is a compiled binary...

**Comment by drowe67** [comment]
> Ok, I'm trying to tickle some brain cells I haven't exercised in a while... Do we need wxrc at all? One problem with cross compiling is that the binaries are built for Windows not Linux. wx-config is a bash script so it works fine but wxrc is a compiled binary...

Good question!  I'm not really sure myself ... 

When you get a moment could you also pls try to help @ra1nb0w with his release/tag request here https://github.com/drowe67/freedv-gui/pull/68 ?

**Comment by hobbes1069** [comment]
Ok, I found the problem but not the solution...

After turning debugging on in the CMake FindwxWidgets module I finally get the error:

`wxWidgets not found due to following missing libraries: wxregexu-3.1-x86_64-w64-mingw32 wxscintilla-3.1-x86_64-w64-mingw32`

Bootstrapping wxWidgets produces the following static libraries:
```
libwx_baseu-3.1-x86_64-w64-mingw32.a
libwx_baseu_net-3.1-x86_64-w64-mingw32.a
libwx_baseu_xml-3.1-x86_64-w64-mingw32.a
libwx_mswu_adv-3.1-x86_64-w64-mingw32.a
libwx_mswu_aui-3.1-x86_64-w64-mingw32.a
libwx_mswu_core-3.1-x86_64-w64-mingw32.a
libwx_mswu_gl-3.1-x86_64-w64-mingw32.a
libwx_mswu_html-3.1-x86_64-w64-mingw32.a
libwx_mswu_media-3.1-x86_64-w64-mingw32.a
libwx_mswu_propgrid-3.1-x86_64-w64-mingw32.a
libwx_mswu_qa-3.1-x86_64-w64-mingw32.a
libwx_mswu_ribbon-3.1-x86_64-w64-mingw32.a
libwx_mswu_richtext-3.1-x86_64-w64-mingw32.a
libwx_mswu_stc-3.1-x86_64-w64-mingw32.a
libwx_mswu_webview-3.1-x86_64-w64-mingw32.a
libwx_mswu_xrc-3.1-x86_64-w64-mingw32.a
libwxregexu-3.1-x86_64-w64-mingw32.a
libwxscintilla-3.1-x86_64-w64-mingw32.a
```

The last two are the "problem" but I can't see why, they seemed to be named exactly as the error shows.

**Comment by hobbes1069** [comment]
https://trac.wxwidgets.org/ticket/18885#ticket

**Comment by drowe67** [comment]
Nice work @hobbes1069 - looks like you are making some progress :+1:

**Comment by hobbes1069** [comment]
Found the root cause but still working on a solution. The problem is the Fedora MinGW CMake toolchain file is specifying CMAKE_FIND_ROOT_PATH which I don't think has ever changed, but CMake is prepending it even if I'm specifying an absolute path to the libraries already.

**Comment by hobbes1069** [comment]
I think I got it fixed, but now I have to re-teach myself to add a commit to a pull request :)

**Comment by drowe67** [comment]
Nice work @hobbes1069 - I'll see if I can get it to build using wxWidgets 3.1 & Docker this weekend!

**Comment by drowe67** [comment]
@hobbes1069 is there a two step process to build freedv-gui when bootstrapping wxWidgets 3.1? 
```
$CMAKE -DCMAKE_BUILD_TYPE=Debug -DCODEC2_BUILD_DIR=$CODEC2DIR/$BUILD_DIR -DLPCNET_BUILD_DIR=$LPCNETDIR/$BUILD_DIR -DBOOTSTRAP_WXWIDGETS=1 ..
make VERBOSE=1
<snip>
make[2]: Leaving directory '/home/build/freedv-gui/build_win64'
fdv_win_fed30_c     | [100%] Built target wxWidgets
fdv_win_fed30_c     | make[1]: Leaving directory '/home/build/freedv-gui/build_win64'
fdv_win_fed30_c     | /usr/bin/cmake -E cmake_progress_start /home/build/freedv-gui/build_win64/CMakeFiles 0
fdv_win_fed30_c     | + '[' mingw64-cmake = mingw64-cmake ']'
fdv_win_fed30_c     | + cd build_win64
fdv_win_fed30_c     | + make package
fdv_win_fed30_c     | + make package
fdv_win_fed30_c     | make: *** No rule to make target 'package'.  Stop.
fdv_win_fed30_c exited with code 2
```

**Comment by hobbes1069** [comment]
Yes, once you complete the first build you just run "cmake ." so it can find the libraries and then make again.

**Pull request merged**